### DISCUSSION TITLE
Prisma returns stale data in both queryRaw and normal query functions

### DISCUSSION THREAD

**Discussion by srinivas-dev-reddy** [prisma/prisma#21026] [Q&A] [answered]
### Bug description

We are using postgresql as database with prisma 4.11.0 in nestjs.

We have discovered an issue where after updating row in Dbeaver the prisma query is returning stale data in both queryRaw and nomal query functions.

Can anyone point me to why this is happening and how to rectify this issue?

PS: this query is running in a cron job

### How to reproduce

1. Create a model
2. Create some rows in the model
3. Update the row data in dbeaver or any other database manager
4. Check the prisma query response
5. It will return the old row data

### Expected behavior

Prisma query should return data from database not from some cache (unless explicitly configured).

### Prisma information

```prisma
model DailyChanges {
  id String @id @default(uuid())

  businessId String

  value   Float
  date          DateTime @db.Date

  orders Int @default(0)

  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  @@unique([businessId, date])
  @@index([businessId, date(sort: Desc)])
}
```

```ts
 this.prisma.dailyChanges.findFirst({
        where: {
          businessId: business.id,
        },
        skip: 1,
        orderBy: {
          date: 'desc',
        },
   })
```

### Environment & setup

- OS: Ubuntu 20.04  
- Database: PostgresSql 
- Node.js version: v16.14.2 

### Prisma Version

```
prisma@4.11.0
```

**Comment by janpio** [comment]
Prisma only talks to the database directly, and does not include a cache out of the box (unless you are using Accelerate, which I assume you are not) - so this is almost impossible a problem with Prisma, but probably for some reason your database returns the old, stale data.

What information does `prisma studio` show?
Are you 100% sure you are using the correct connection string in Prisma?
Do other Node based database clients with the same connection string also show stale data?

**Reply by srinivas-dev-reddy** [reply]
- When i query data using dbeaver it shows correct data but in prisma the returned data is stale
- Regarding the connection string - yes we are using the correct connection string

**Comment by mustofa-id** [comment]
DBeaver has a manual commit feature when labeling a connection as 'production.' Are you sure you've committed the changes?

**Reply by srinivas-dev-reddy** [reply]
Yes the changes are commited.

**Comment by srinivas-dev-reddy** [comment, marked as answer]
I think i figured out the issue , it seems that the postgres connection to server was timing out but prisma was not throwing them when i removed other connections to postgres and restarted the server. Everything works fine now.

**Reply by janpio** [reply]
Did Prisma do something wrong? When a connection is dead, Prisma will definitely also not show stale data.
